mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -3020,6 +3044,7 @@ void InsertOp::getCanonicalizationPatterns(RewritePatternSet &results,
                                            MLIRContext *context) {
   results.add<InsertToBroadcast, BroadcastFolder, InsertSplatToSplat,
               InsertOpConstantFolder>(context);
+  results.add(foldPoisonIndexInsertExtractOp<InsertOp>);
```

Comment:
You can make this a folder instead of a canonicalization pattern. You can just return ub::PoisonAttr and the ub dialect will materialize the poison attr if not used. Example:

https://github.com/llvm/llvm-project/blob/43a50deb63453cd3c800f097514d500536f9d436/mlir/lib/Dialect/Affine/IR/AffineOps.cpp#L620

Reply:
Thanks! I added it to both as we don't want poison indices to be part of the canonical form

Reply:
Interesting. My understanding of folders is that they are canonicalization themselves, just more restricted and local. I don't think anything that is already in the folder should go into the canonicalization.

(I don't want to block anything, if you think this should be here, i'm fine with it, i just think it might be overkill)